### ISSUE TITLE
Implement more traits for UriRefBuf and UriBuf

### ISSUE THREAD

**Issue by Luro02** [google/rust-async-coap#22]
The following traits could be implemented
- [`FromStr`](https://doc.rust-lang.org/std/str/trait.FromStr.html)
- [`TryFrom`](https://doc.rust-lang.org/std/convert/trait.TryFrom.html)
-> `TryFrom<&str>`
-> `TryFrom<&String>`
-> `TryFrom<String>` (only for `UriBuf`)
-> `impl<T: AsRef<UriRef>> TryFrom<T>`

Most of these traits can be simply delegated to their respective function

```rust
use std::str::FromStr;

impl FromStr for UriBuf {
   type Err = ParseError;

    fn from_str(input: &str) -> Result<Self, Self::Err> { Self::from_str(input) }
}
```

**Comment by Luro02** [comment]
I am working on this :)

**Comment by darconeous** [comment]
Cool! Having a look now.

**Comment by darconeous** [comment]
This landed via 99a61eee26d9bf9.
